### PR TITLE
Make backupMap volatile to make it visible to user thread

### PR DESCRIPTION
**Pull request by ahmetmircik**
fixes http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/2378/testReport/junit/com.hazelcast.splitbrainprotection.queue/QueueSplitBrainProtectionReadTest/getLocalQueueStats_splitBrainProtection_splitBrainProtectionType_READ_/

```
java.lang.NullPointerException
	at com.hazelcast.collection.impl.queue.QueueContainer.createLinkedList(QueueContainer.java:936)
	at com.hazelcast.collection.impl.queue.QueueContainer.getItemQueue(QueueContainer.java:914)
	at com.hazelcast.collection.impl.queue.QueueContainer.size(QueueContainer.java:654)
	at com.hazelcast.collection.impl.queue.QueueService.createLocalQueueStats(QueueService.java:348)
	at com.hazelcast.collection.impl.queue.QueueProxyImpl.getLocalQueueStats(QueueProxyImpl.java:51)
	at com.hazelcast.splitbrainprotection.queue.QueueSplitBrainProtectionReadTest.getLocalQueueStats_splitBrainProtection(QueueSplitBrainProtectionReadTest.java:91)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java [hunk 1]
```diff
@@ -99,6 +98,8 @@
     // to avoid reloading same items
     private long lastIdLoaded;
 
+    private volatile Map<Long, QueueItem> backupMap;
```
Comment (petrpleshachkov) [inline hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java:101]: The `backupMap` is initialized with the `HashMap` that is not thread-safe. Is it handled properly that multiple threads are not accessing it?

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java]: fixed it by making `backupMap` a CHM.

**Review by petrpleshachkov — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java [hunk 2]
```diff
@@ -980,24 +986,29 @@ public boolean hasEnoughCapacity(int delta) {
      * @return backup replica map from item ID to queue item
      */
     public Map<Long, QueueItem> getBackupMap() {
+        ConcurrentMap<Long, QueueItem> backupMap = this.backupMap;
         if (backupMap == null) {
+            Queue<QueueItem> itemQueue = this.itemQueue;
```
Comment (mmedenjak) [inline hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java:991]: This mix of referring to a local variable and the field is a bit confusing. Can you elaborate a bit or simply use the local variable only when checking the size and adding to backup map?

**Review by mmedenjak — COMMENTED** [review]
Overall looks good, only confusion is around handling of `itemQueue`

File: hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java (hunk 2 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java]: now it should be more clear

**Review by mmedenjak — APPROVED** [review]

**Pull request merged**